### PR TITLE
feat(estree): Add definitions for StaticBlock statement

### PR DESCRIPTION
**Pull request by sanex3339**
ESTree: add support for class StaticBlock statement

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/estree/estree/blob/master/es2022.md#staticblock

### CODE REVIEW

File: types/estree/index.d.ts [hunk 1]
```diff
@@ -104,6 +104,10 @@ export interface BlockStatement extends BaseStatement {
   innerComments?: Array<Comment> | undefined;
 }
 
+export interface StaticBlock extends Omit<BlockStatement, 'type'> {
```
Comment (sanex3339) [inline types/estree/index.d.ts:107]: As it described here:
https://github.com/estree/estree/blob/master/es2022.md#staticblock

Comment (amcasey) [reply types/estree/index.d.ts:107]: Alternatively, you could introduce a `BaseBlock` type, eliminating the need for `Omit`.

**Review by amcasey — COMMENTED** [review]
I guess this package isn't versioned since there's no backing library?

The code looks correct, but I'd like to hear from an owner, since this package is Critical.

**Comment by sanex3339** [comment]
> Alternatively, you could introduce a BaseBlock type, eliminating the need for Omit.

I'd prefer to keep it as is right now because it's the only such case.
In the future, if we have more similar cases, we can extract some fields to some generic `BaseStatement` but for now, with only a single case I'm not sure if such extraction makes sense.

**Comment by sanex3339** [comment]
Hi, @RReverser, can you please take a look at it? It's not urgent

**Review by RReverser — APPROVED** [review]
I agree with the `BaseBlock` nit, but otherwise looks reasonable.

**Comment by sanex3339** [comment]
I'm going to merge it as is to follow the es2022 spec that says that `StaticBlock` should be extended from `BlockStatement`

**Comment by sanex3339** [comment]
Ready to merge

**Pull request merged**